app/controllers/application_controller.rb

```diff
@@ -7,4 +8,122 @@ class ApplicationController < Sinatra::Base
     set :views, 'app/views'
   end
 
+  enable :sessions
+
+  get '/tweets/new' do 
+    redirect :'/login' if !session[:id]
+    erb "<form action='/tweets/new' method='post'>
+    <input type='text' name='content'>Tweet</input>
+    <input type='submit' id='submit'>
+    </form>"
+  end
+
+  post '/tweets/new' do 
+    redirect :'/tweets/new' if params[:content].blank?
+    user = User.find(session[:id])
+    user.tweets.build(params)
+    user.save
+  end
+
+  get '/' do
+    erb 'Welcome to Fwitter'
+  end
+
+  get '/signup' do 
+    redirect :'/tweets' if session[:id]
+  end
+
+  post '/signup' do
+    redirect :'/signup' if params[:username]==""
+    redirect :'/signup' if params[:email]=="" 
+    redirect :'/signup' if params[:password]==""
+    
+    user = User.create(params)
+    session[:id] = user.id
+
+    redirect :'/tweets'
+  end
+
+  get '/login' do
+
+    redirect :'/tweets' if session[:id]
+    
+    erb "<%= session[:id].nil? %>
+    <form action='/login' method='post'>
+    <input type='text' name='username'>Username</input>
+    <input type='text' name='password'>Password</input>
+    <input type='submit' id='submit'>
+    </form>"
+
+    
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username] ).try(:authenticate, params[:password])
+    session[:id] = user.id
+    redirect :'/tweets'
+  end
+
+  get '/tweets' do 
+    redirect :'/login' if !session[:id]
+    erb "Welcome,
+      <%= Tweet.all.each {|tweet| tweet.content} %>"
+  end
+
+  get '/logout' do
+    session.clear
+    redirect :'/login'
+  end
+
+  get "/users/:slug" do 
+    @user = User.find_by_slug(params[:slug])
+    erb "
+      <%= @user.tweets.each {|tweet| tweet.content} %> 
+    "
+  end
+
+  get "/tweets/:id/edit" do 
+    redirect :'/login' if !session[:id]
+    @tweet = Tweet.find(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?